frontend/analysis_options.yaml

```diff
@@ -1,13 +1,164 @@
+    no_runtimeType_toString: error
+    non_constant_identifier_names: error
+    noop_primitive_operations: error
+    null_check_on_nullable_type_parameter: error
+    null_closures: error
+    overridden_fields: error
+    package_names: error
+    package_prefixed_library_names: error
+    parameter_assignments: error
+    prefer_asserts_in_initializer_lists: error
+    prefer_collection_literals: error
+    prefer_conditional_assignment: error
+    prefer_const_constructors: error
+    prefer_const_constructors_in_immutables: error
+    prefer_const_declarations: error
+    prefer_const_literals_to_create_immutables: error
+    prefer_constructors_over_static_methods: error
+    prefer_contains: error
+    prefer_equal_for_default_values: error
+    prefer_final_fields: error
+    prefer_final_in_for_each: error
+    prefer_final_locals: error
+    prefer_for_elements_to_map_fromIterable: error
+    prefer_function_declarations_over_variables: error
+    prefer_generic_function_type_aliases: error
+    prefer_if_elements_to_conditional_expressions: error
+    prefer_if_null_operators: error
+    prefer_initializing_formals: error
+    prefer_inlined_adds: error
+    prefer_interpolation_to_compose_strings: error
+    prefer_is_empty: error
+    prefer_is_not_empty: error
+    prefer_is_not_operator: error
+    prefer_iterable_whereType: error
+    prefer_null_aware_method_calls: error
+    prefer_spread_collections: error
+    prefer_typing_uninitialized_variables: error
+    prefer_void_to_null: error
+    provide_deprecation_message: error
+    recursive_getters: error
+    require_trailing_commas: error
+    sized_box_for_whitespace: error
+    slash_for_doc_comments: error
+    sort_child_properties_last: error
+    sort_unnamed_constructors_first: error
+    test_types_in_equals: error
+    throw_in_finally: error
+    tighten_type_of_initializing_formals: error
+    type_annotate_public_apis: error
+    type_init_formals: error
+    unnecessary_await_in_return: error
+    unnecessary_brace_in_string_interps: error
+    unnecessary_const: error
+    unnecessary_getters_setters: error
+    unnecessary_new: error
+    unnecessary_null_aware_assignments: error
+    unnecessary_null_checks: error
+    unnecessary_null_in_if_null_operators: error
+    unnecessary_nullable_for_final_variable_declarations: error
+    unnecessary_overrides: error
+    unnecessary_parenthesis: error
+    unnecessary_raw_strings: error
+    unnecessary_statements: error
+    unnecessary_string_escapes: error
+    unnecessary_string_interpolations: error
+    unnecessary_this: error
+    unrelated_type_equality_checks: error
+    unsafe_html: error
+    use_full_hex_values_for_flutter_colors: error
+    use_function_type_syntax_for_parameters: error
+    use_named_constants: error
+    use_setters_to_change_properties: error
+    use_string_buffers: error
+    use_test_throws_matchers: error
+    valid_regexps: error
+    void_checks: error
```

Comment:
These are a lot of custom rules. Why are they not part of https://github.com/passsy/dart-lint

How much difference would you expect if we stick with https://pub.dev/packages/flutter_lints and add again more pedantic rules? I would like to avoid third-party dependencies as much as possible.

Reply:
That are not custom rules but I rather just set them to error. All those rules are enabled in lint as well, but perhaps with just warning or info severity. Actually I copied those from lint.

Reply:
I'd rather use a package with recommended rules and wide adoption across other projects then setting up lots of custom rules. `flutter_lints` was completely enabled before already, so those rules are very (too) loose imo.

Reply:
200 starts does not mean it is adopted across projects. I think this is just a dependency which could cause problems in the future. While just copy pasting the rules into one file should work correctly.

Reply:
What kind of problems do you think there could be?

Reply:
Any non-offical dependency can become outdated and unmaintained. We have several of such packages in the integreat project. If it is not really required to have a dependency then it is better to avoid it.

So just copy pasting the rules over could be more beneficial over the time.

Reply:
Lets just paste it over once we run into problems.